docs/source/tracking/data-api.rst

```diff
@@ -0,0 +1,227 @@
+MLflow Data Module Tutorial
+===========================
+
+The ``mlflow.data`` module is an integral part of the MLflow ecosystem, designed to enhance your machine learning workflow.
+This module enables you to record and retrieve dataset information during model training and evaluation, leveraging MLflow's tracking capabilities.
+
+Key Interfaces
+--------------
+
+There are two main abstract components associated with the ``mlflow.data`` module:
+
+Dataset 
+^^^^^^^
+
+The ``Dataset`` abstraction is a metadata tracking object that holds the information about a given logged dataset.
+
+The information stored within a ``Dataset`` object includes features, targets, and predictions, along with 
+metadata like the dataset's name, digest (hash), schema, and profile. You can log this metadata using the :py:func:`mlflow.log_input` API. 
+The module provides functions to construct :py:class:`mlflow.data.dataset.Dataset` objects from various data types.
+
+There are a number of concrete implementations of this abstract class, including:
+
+- :py:class:`mlflow.data.spark_dataset.SparkDataset`
+- :py:class:`mlflow.data.pandas_dataset.PandasDataset`
+- :py:class:`mlflow.data.numpy_dataset.NumpyDataset`
+- :py:class:`mlflow.data.huggingface_dataset.HuggingFaceDataset`
+- :py:class:`mlflow.data.spark_dataset.SparkDataset`
+- :py:class:`mlflow.data.tensorflow_dataset.TensorFlowDataset`
+
+The following example demonstrates how to construct a :py:class:`mlflow.data.pandas_dataset.PandasDataset` object from a Pandas DataFrame:
+
+.. code-block:: python
+
+    import mlflow.data
+    import pandas as pd
+    from mlflow.data.pandas_dataset import PandasDataset
+
+
+    CSV_DELIMITER = ";"
```

Comment:
nit: do we need to declare variable?

Reply:
no, not really. It was in the interest of keeping me from forgetting to declare the separator while I was building the examples ;) removed it and used the static definition instead